lib/facility.rb

```diff
@@ -1,14 +1,82 @@
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_reader :name, 
+              :address, 
+              :phone, 
+              :services, 
+              :registered_vehicles, 
+              :collected_fees
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  def initialize(facility_details)
+    @name = facility_details[:name]
+    @address = facility_details[:address]
+    @phone = facility_details[:phone]
     @services = []
+    @registered_vehicles = []
+    @collected_fees = 0
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
-end
+
+  def registration_helper(vehicle)
+    return if !@services.include?('Vehicle Registration')
+    vehicle.registration_date = Date.today
+    @registered_vehicles << vehicle
+  end
+
+  def register_ev(vehicle)
+    return if !@services.include?('Vehicle Registration')
+    vehicle.plate_type = :ev
+    @collected_fees += 200
+    registration_helper(vehicle)
+  end
+
+  def register_antique(vehicle)
+    return if !@services.include?('Vehicle Registration')
+    vehicle.plate_type = :antique
+    @collected_fees += 25
+    registration_helper(vehicle)
+  end
+
+  def register_reg(vehicle)
+    return if !@services.include?('Vehicle Registration')
+    vehicle.plate_type = :regular
+    @collected_fees += 100
+    registration_helper(vehicle)
+  end
```

Comment:
I LOVE that you are creating helper methods to assist in registering specific vehicle types. This is really great.